### PR TITLE
Remove any restored tab that cannot be loaded.

### LINKED ISSUE #1216: Should not attempt to restore a folder on an unmounted device on startup
# Prerequisites
- [x] I have searched open and closed issues for duplicates.

# Describe the error

Opening the Files application after having unmounted an external device makes it attempt to navigate to the now-unmounted device, which it instead displays as a non-existent folder.

## To reproduce

1. Open Files.
2. Mount an external device such as a USB drive.
3. Close Files.
4. Physically remove the external device.
5. Open Files.
6. The application attempts to navigate to a location which does not exist, and a “this directory does not exist” message is shown.

## Expected behaviour

The application does not try to re-open a folder if it does not exist, regardless of whether it is an unmounted device or some other type of non-existent path. If all folders from the previous session no longer exist, the application opens the Home folder.

## Screenshots or screen recordings

[Screen recording](https://github.com/elementary/files/files/4144546/Opening.the.application.after.unmounting.an.external.device.makes.it.navigate.to.a.non-existent.folder.zip)

# Platform information

System information 2

- [ ] I'm using the latest version from git that I've manually compiled
- [x] I'm using the latest released stable version

Language: French
Keyboard: English (US) - English (Dvorak intl., with dead keys)
Location: Paris

The testing system is installed on a small partition on a ThinkPad T470 laptop on which UEFI Secure Boot is disabled. Aside from specifying the language, keyboard, location, initial user, and the specific partition in which to install elementary OS, all of the default settings were used during the installation. No Internet connection was used during the installation. The only system-wide changes which have been made after the installation are that the system connects to the Internet by wi-fi via a Linksys WRT54GL Wireless Router and that [Screen Recorder](https://appcenter.elementary.io/com.github.mohelm97.screenrecorder/) has been installed via AppCentre.

# Additional context

Before this test was performed, the following steps were followed:

1. All system-wide changes from previous tests were reverted.
2. All system components were updated via the blue “Update All” button in AppCentre.
3. A new administrator user named “Essai” or “Essai 2” was created with the password “komalleana”.
4. The current user was logged out and the new user was logged in.
5. The old user was deleted, the screen resolution was set to 800x600 at 60 Hz, tap clicking was disabled, and the touchpad pointer speed was set to the maximal value.
6. The computer was rebooted.
6. The test user was logged in.

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #1216 

* Refactor removing a tab
* Better function names

### CODE REVIEW

File: src/View/Window.vala [hunk 1]
```diff
@@ -1096,15 +1098,15 @@ namespace Marlin.View {
             debug ("Mount %s removed", mount.get_name ());
             GLib.File root = mount.get_root ();
 
-            foreach (var page in tabs.get_children ()) {
-                var view_container = page as Marlin.View.ViewContainer;
+            foreach (var tab in tabs.tabs) {
```
Comment (tintou) [inline src/View/Window.vala]:
```suggestion
            foreach (unowned Granite.Widgets.Tab tab in tabs.tabs) {
```

File: src/View/Window.vala [hunk 2]
```diff
@@ -1096,15 +1098,15 @@ namespace Marlin.View {
...
+                var view_container = tab.page as Marlin.View.ViewContainer;
```
Comment (tintou) [inline src/View/Window.vala]:
```suggestion
                unowned Marlin.View.ViewContainer view_container = (Marlin.View.ViewContainer) tab.page;
```

File: src/View/Window.vala (hunk 1 above)
Comment (jeremypw) [reply src/View/Window.vala]: I'm always forgetting that...

**Comment by jeremypw** [comment]
Blocking as some objections have been raised to the original issue.  Awaiting UX adjudication.

**Comment by jeremypw** [comment]
Closing pending design team adjudication on original issue.

**Pull request closed without merging**